# Part 1

In [18]:
import pandas as pd

# First choice: dataset 1

In [19]:
df1 = pd.read_csv('/content/Metabolic  Syndrome.csv')
df1.head()

seqn  Age     Sex  Marital  Income   Race  WaistCirc   BMI  Albuminuria  \
0  62161   22    Male   Single  8200.0  White       81.0  23.3            0   
1  62164   44  Female  Married  4500.0  White       80.1  23.2            0   
2  62169   21    Male   Single   800.0  Asian       69.6  20.1            0   
3  62172   43  Female   Single  2000.0  Black      120.4  33.3            0   
4  62177   51    Male  Married     NaN  Asian       81.1  20.1            0   

   UrAlbCr  UricAcid  BloodGlucose  HDL  Triglycerides MetabolicSyndrome  
0     3.88       4.9            92   41             84         No MetSyn  
1     8.55       4.5            82   28             56         No MetSyn  
2     5.07       5.4           107   43             78         No MetSyn  
3     5.22       5.0           104   73            141         No MetSyn  
4     8.13       5.0            95   43            126         No MetSyn

In [20]:
df1.columns

Index(['seqn', 'Age', 'Sex', 'Marital', 'Income', 'Race', 'WaistCirc', 'BMI',
       'Albuminuria', 'UrAlbCr', 'UricAcid', 'BloodGlucose', 'HDL',
       'Triglycerides', 'MetabolicSyndrome'],
      dtype='object')

The target of this dataset is  'MetabolicSyndrome' column

In [21]:
df1.shape

(2401, 15)

In [22]:
df1.dtypes

seqn                   int64
Age                    int64
Sex                   object
Marital               object
Income               float64
Race                  object
WaistCirc            float64
BMI                  float64
Albuminuria            int64
UrAlbCr              float64
UricAcid             float64
BloodGlucose           int64
HDL                    int64
Triglycerides          int64
MetabolicSyndrome     object
dtype: object

In [23]:
df1.isnull().sum()

seqn                   0
Age                    0
Sex                    0
Marital              208
Income               117
Race                   0
WaistCirc             85
BMI                   26
Albuminuria            0
UrAlbCr                0
UricAcid               0
BloodGlucose           0
HDL                    0
Triglycerides          0
MetabolicSyndrome      0
dtype: int64

## 1.Data Source

[Metabolic Syndrome Prediction - dataset by informatics-edu](https://)
The dataset come from NHANES National Health and Nutrition Examination Survey) , created 5 years ago by  
@rhoyt (Robert Hoyt: Author and Editor of Health Informatic)


## 2. Data Description

The Metabolic Syndrome Prediction Data set was combined from multiple tables with SQL: abnormal waist circumference, triglycerides above 150, HDL cholesterol below 50 in women or 40 in men, history of hypertension and mildly elevated fasting blood sugar (100-125). uric acid, race, income were added that might contribute the model.

## 3. What Does a row represent?

In [24]:
df1.head(3)

seqn  Age     Sex  Marital  Income   Race  WaistCirc   BMI  Albuminuria  \
0  62161   22    Male   Single  8200.0  White       81.0  23.3            0   
1  62164   44  Female  Married  4500.0  White       80.1  23.2            0   
2  62169   21    Male   Single   800.0  Asian       69.6  20.1            0   

   UrAlbCr  UricAcid  BloodGlucose  HDL  Triglycerides MetabolicSyndrome  
0     3.88       4.9            92   41             84         No MetSyn  
1     8.55       4.5            82   28             56         No MetSyn  
2     5.07       5.4           107   43             78         No MetSyn

A row represents a person with informations about him such as: Sex, Marital Status, Income, Race ...

## 4. Classification / Regression Problem

In [25]:
df1['MetabolicSyndrome'].value_counts()

MetabolicSyndrome
No MetSyn    1579
MetSyn        822
Name: count, dtype: int64

It is about a classification problem as we have to predict a class  (No MetSyn, MetSyn) which is a categorical variable.  

## 5.  How many features does the data have?

In [26]:
df1.shape

(2401, 15)

The Data Have 15 Features

## 6.How many rows are in the dataset?

In [27]:
df1.shape

(2401, 15)

The Dataset have 2401 rows.

## 7. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

In [28]:
df1.head(10)

seqn  Age     Sex   Marital  Income   Race  WaistCirc   BMI  Albuminuria  \
0  62161   22    Male    Single  8200.0  White       81.0  23.3            0   
1  62164   44  Female   Married  4500.0  White       80.1  23.2            0   
2  62169   21    Male    Single   800.0  Asian       69.6  20.1            0   
3  62172   43  Female    Single  2000.0  Black      120.4  33.3            0   
4  62177   51    Male   Married     NaN  Asian       81.1  20.1            0   
5  62178   80    Male   Widowed   300.0  White      112.5  28.5            0   
6  62184   26    Male    Single  9000.0  Black       78.6  22.1            0   
7  62189   30  Female   Married  6200.0  Asian       80.2  22.4            0   
8  62191   70    Male  Divorced  1000.0  Black        NaN   NaN            1   
9  62195   35    Male       NaN  2500.0  Black       99.0  28.2            0   

   UrAlbCr  UricAcid  BloodGlucose  HDL  Triglycerides MetabolicSyndrome  
0     3.88       4.9            92   41             84         No MetSyn  
1     8.55       4.5            82   28             56         No MetSyn  
2     5.07       5.4           107   43             78         No MetSyn  
3     5.22       5.0           104   73            141         No MetSyn  
4     8.13       5.0            95   43            126         No MetSyn  
5     9.79       4.8           105   47            100         No MetSyn  
6     9.21       5.4            87   61             40         No MetSyn  
7     8.78       6.7            83   48             91         No MetSyn  
8    45.67       5.4            96   35             75         No MetSyn  
9     2.21       6.7            94   46             86         No MetSyn

Cleaning Data: Missing Values

# Second choice: dataset 2

In [37]:
df2 = pd.read_csv('/content/Mall_Customers[1].csv')
df2.head()

CustomerID   Genre  Age  Annual Income (k$)  Spending Score (1-100)
0           1    Male   19                  15                      39
1           2    Male   21                  15                      81
2           3  Female   20                  16                       6
3           4  Female   23                  16                      77
4           5  Female   31                  17                      40

## 1. Source

[https://www.kaggle.com/datasets/shwetabh123/mall-customers](https://)
 The Source of our Dataset is Kaggle

## 2. Data Description

The Dataset is downloaded from Kaggle, it has no description but is about Mall customers data. The target Column is the spending score of customers so we have to predict this value by modeling.

## 3. The Target

In [30]:
df2.head(5)

CustomerID   Genre  Age  Annual Income (k$)  Spending Score (1-100)
0           1    Male   19                  15                      39
1           2    Male   21                  15                      81
2           3  Female   20                  16                       6
3           4  Female   23                  16                      77
4           5  Female   31                  17                      40

In [31]:
df2.columns

Index(['CustomerID', 'Genre', 'Age', 'Annual Income (k$)',
       'Spending Score (1-100)'],
      dtype='object')

In [32]:
Y= df2['Spending Score (1-100)']

The target is 'Spending Score (1-100)' column.

## 4. What does one row represent?

A row represent a person with information: sex, age, income ...

## 5. Classification / Regression Problem:

It is about a regression problem as the model has to predict a score (numeric variable)

## 6. How many features does the data have?

In [38]:
df2.shape

(200, 5)

The data has 5 features

## 7. How many rows are in the dataset?

The Data 200 rows

## 8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

The challenge with this dataset is the volume. The number of rows and features which can be not enough for modeling.